### DISCUSSION TITLE
make a redirection before rendering static path with catch all

### DISCUSSION THREAD

**Discussion by svassaux** [vercel/next.js#19575] [Help]
Hello everybody!

I have a site with statically rendered pages. It works :)

For that purpose, under `pages` I have a catch all [...slug] containing an index.js file with getStaticProps and getStaticPaths methods.

However, I would like to redirect some urls to some of those static pages.

If a user gets to the `baseurl/non-existing-page-id` then I would like to redirect this user to a specific static page.
I am struggling to find a way to do so. My ideas so far:

- find a way to keep a getServerSideProps handle the slug if the page is not within getStaticPaths
- have the fallback accept a method to define what happens if the path is not found

Another need: I might want to redirect some slugs to an external website. Example, if a user visits baseurl/login, I want to redirect this user to app.baseurl/login while keeping static generation for my site.

Does someone see a way to solve this redirection issue ?

**Comment by michrome** [comment]
Hey @svassaux could you do this with [Wildcard Path Matching](https://nextjs.org/docs/api-reference/next.config.js/redirects#wildcard-path-matching)?

**Reply by svassaux** [reply]
I don't see how the wildward would help but the article about redirections is interesting.

I could try to find a way to generate a list of redirects at build time and have it hydrate next.config.js

I'll git it a try! thanks :)
